test/LPIncentiveHook.t.sol

```diff
@@ -1346,6 +1346,227 @@ contract LPIncentiveHookTest is Test, Deployers {
         );
     }
 
+    function test_RedeemRewards() public {
+        deal(Currency.unwrap(rewardToken), address(hook), 1e52);
+        // Deal tokens to alice
+        deal(Currency.unwrap(token0), alice, 100000 ether);
+        deal(Currency.unwrap(token1), alice, 100000 ether);
+
+        // Approve tokens for router
+        vm.startPrank(alice);
+        IERC20(Currency.unwrap(token0)).approve(address(modifyLiquidityRouters[alice]), type(uint256).max);
+        IERC20(Currency.unwrap(token1)).approve(address(modifyLiquidityRouters[alice]), type(uint256).max);
+        vm.stopPrank();
+
+        // Create test params
+        int24 tickLower = -120;
+        int24 tickUpper = 120;
+        uint256 liquidity = 1000e18;
+
+        // Add liquidity
+        addLiquidity(alice, liquidity, tickLower, tickUpper);
+
+        // Wait some time for rewards to accumulate
+        advanceTime(1000);
+
+        // Remove liquidity
+        removeLiquidity(alice, liquidity, tickLower, tickUpper);
+
+        // Get accumulated rewards
+        uint256 aliceRewards = hook.accumulatedRewards(address(modifyLiquidityRouters[alice]));
+        assertGt(aliceRewards, 0, "Alice should have rewards");
+
+        // Record contract balance before redemption
+        uint256 contractBalanceBefore = IERC20(Currency.unwrap(rewardToken)).balanceOf(address(hook));
+        uint256 aliceBalanceBefore =
+            IERC20(Currency.unwrap(rewardToken)).balanceOf(address(modifyLiquidityRouters[alice]));
+
+        // Claim rewards
+        vm.prank(address(modifyLiquidityRouters[alice]));
+        hook.redeemRewards();
```

Comment:
I guess it is linked to the fact that we don't have an easy access to sender (alice) instead of the router. But for this test/feature, I would assume that this needs to be sent from Alice's account, and not her router.

In that case, i'm not sure how this would be possible - the router will need to support this.

So I suggest:

1. If it is a simple "fix" given the current structure (no access to the real sender) -> add a comment to explain this situation

or

2. Explain how this will be possible in "prod" so that the router can claim rewrads (i.e. call a custom hook function)